h/views/activity.py

```diff
@@ -90,27 +92,25 @@ def user_link(userid):
         }
 
 
-@view_defaults(route_name='activity.group_search',
-               renderer='h:templates/activity/search.html.jinja2')
+@view_defaults(route_name='group_read',
+               renderer='h:templates/activity/search.html.jinja2',
+               custom_predicates=[custom_predicates.has_read_permission,
+                                  custom_predicates.feature('search_page')])
```

Comment:
We only have to state this once, and it means that _all_ view methods in this class will only be called if the user is a member of the group ("has read permission" on the context object, which is the group) and the search_page feature flag is on.